### ISSUE TITLE
Bug 54108923: [React Native Gallery -> Radio Buttons]: Screen reader is not announcing the associated group name and selection state of the radio button.

### ISSUE THREAD

**Issue by Yajur-Grover** [microsoft/react-native-gallery#496]
Build Details:
OS Version: Dev (OS Build 27695.1000)
App: React Native Gallery 
App Version: 1.0.18.0
React Native Windows Version: 0.75.0
AT: Screen Reader
 
Pre-Requisite:
Turn on Narrator (Ctrl + Win + Enter).
 
Repro-Steps:
Launch the React Native Gallery app.
Navigate to ‘All samples’ button present in left navigation pane and invoke it.
Navigate to ‘Radio Buttons’ button present under “Basic Input” group and invoke it.
Navigate to radio buttons and listen to the narrator's announcement.
Actual Result:
Screen reader is not announcing the associated group name and selection state of the radio button. When user navigated to the radio buttons, narrator is simply announcing the information about the radio button and it not announcing the label name and state of the button.

Expected Result:
Label “Options” should be associated with radio group and screen reader should announce the selection state of the radio button. When user navigates to the radio button, narrator should announce as "Options, Option 1 radio button, 1 of 3, selected".
 
User Impact:
Users with visual impairment who rely on screen reader will not be able to know the label while navigating to the radio group if screen reader does not announce that.
 
Link to ADO bug: https://dev.azure.com/microsoft/OS/_workitems/edit/54108923

**Comment by chrisglein** [comment]
Role is set but isn't creating the right announce behavior around the group/parent. May be a platform issue assuming everything is tagged correctly.
